### PR TITLE
Stores value as numeric for number input types

### LINKED ISSUE #18: Storing number types
I'm finding that when I use the input type=number element I get strings instead of numbers.

I want to propose a change to [at part of the code](https://github.com/Polymer/NodeBind/blob/master/src/NodeBind.js#L175-L180)

to include this:

```js
if (input.type==='number')
  observable.setValue(input.valueAsNumber);
else
  observable.setValue(input[property]);
```

This then stores the user input as a number type

### PR DESCRIPTION
**Pull request by pflannery**
added numeric test

closes #18

### CODE REVIEW

File: src/NodeBind.js [hunk 1]
```diff
@@ -173,7 +173,11 @@
     var eventType = getEventForInputType(input);
 
     function eventHandler() {
-      observable.setValue(input[property]);
+      if (property == 'value' && input.type == 'number' )
+        observable.setValue(input.valueAsNumber);
+      else
```
Comment (jmesserly) [inline src/NodeBind.js]: trivial: I don't think the JS style in this file uses dangling else. I'd suggest curlies for the if-else, or maybe restructure:

```js
var isNum = property == 'value' && input.type == 'number';
observable.setValue(isNum ? input.valueAsNumber : input[property]);
```

the JS style guide endorses ternary op for what it's worth: https://google-styleguide.googlecode.com/svn/trunk/javascriptguide.xml?showone=Tips_and_Tricks#Tips_and_Tricks

File: tests/tests.js [hunk 2]
```diff
@@ -652,6 +652,22 @@ suite('Form Element Bindings', function() {
     radioInputCheckedMultipleForms(shadowRoot, done);
   });
 
+  test('(Number)Input.number', function(done) {
```
Comment (jmesserly) [inline tests/tests.js]: given the other `(Radio)Input.checked` test, maybe this should be `(Number)Input.value` for consistency?

**Comment by jmesserly** [comment]
a couple of very minor comments, otherwise LGTM :)

**Comment by jmesserly** [comment]
(and thank you for putting together the pull request!)

**Comment by pflannery** [comment]
@jmesserly thanks, I've responded to your feedback.

**Comment by jmesserly** [comment]
awesome, thanks again

**Pull request merged**